src/clarity/types/tupleCV.ts

```diff
@@ -1,18 +1,32 @@
 import { ClarityType, ClarityValue } from '../clarityValue';
 import { isClarityName } from '../../utils';
 
+type TupleData = { [key: string]: ClarityValue };
+
 interface TupleCV {
   type: ClarityType.Tuple;
-  data: { [key: string]: ClarityValue };
+  data: TupleData;
 }
 
-function tupleCV(data: { [key: string]: ClarityValue }): TupleCV {
+function tupleCV(data: TupleData): TupleCV {
   for (const key in data) {
     if (!isClarityName(key)) {
       throw new Error(`"${key}" is not a valid Clarity name`);
     }
   }
-  return { type: ClarityType.Tuple, data };
+
+  const ordered: TupleData = {};
```

Comment:
I'm not sure this will always be ordered as intended https://stackoverflow.com/questions/5525795/does-javascript-guarantee-object-property-order

I think the previous code `const lexicographicOrder = ....` should be used

Reply:
ah good point. the reason I moved this code was so that I could simply iterate through a tuple's values when trying to match it to an abi, rather than having to search through them. But I'm realizing was based an the assumption that the abi tuple definition will be lexicographically ordered, which I don't think is the case (I think this is only true for serialized tuples). Will move this code back, and implement improved tuple matching.